### PR TITLE
fix: updated script for better api management; added new locations

### PR DESCRIPTION
**Pull request by mousemke**
+ The api has changed and is much quicker kicking people off so i have slowed the query rate and added an offset so they dont fire all at once.

+ I added a bunch more places with vaccinations

+ i changed a bit of the logging to be clearer. when it was failing it was just flooding the console with not so useful info

### CODE REVIEW

**Comment by mousemke** [comment]
i like the idea of them all being unchecked by default so it's really just install and run, but LMK of i should comment them out

File: demon.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ function log(...msg) {
 }
 
 function error(msg) {
-  console.error(new Date().toISOString(), msg);
+  console.error(new Date().toISOString(), msg.errno, msg.code, msg?.config?.url);
```
Comment (Pita) [inline demon.js:49]: nice!

File: demon.js [hunk 2]
```diff
@@ -105,20 +105,22 @@ function notify() {
   });
 
   player.play("./bell-ring-01.wav", function (err) {
-    if (error) {
+    if (err) {
```
Comment (Pita) [inline demon.js:137]: good catch!

File: demon.js [hunk 3]
```diff
@@ -198,39 +201,69 @@ const data = [
...
+    xhrLink: "https://www.doctolib.de/availabilities.json?start_date=2021-05-30&visit_motive_ids=2898162&agenda_ids=469719&insurance_sector=public&practice_ids=162056&destroy_temporary=true&limit=4"
+  }
 ];
 
-data.forEach(function (links) {
-  observe(links.xhrLink, links.bookingLink);
+let offset = 0;
+data.forEach((links) => {
+  offset = offset + 100;
+  observe(links.xhrLink, links.bookingLink, offset);
```
Comment (Pita) [inline demon.js]: The third parameter of observe is `secondShotXhrLink`, not `offset`. But I think we can safely remove that feature, as its only implemented for one Doctor

**Review by Pita — CHANGES_REQUESTED** [review]
Thank you for these changes! It's totaly ok to have all options enabled by default, we just need to change the README then to reflect this behauvior.

File: demon.js (hunk 3 above)
Comment (mousemke) [reply demon.js]: i went through and commented some things.  i think there is some second shot availability links that we just cant get the lint to unless we get there on the page with the inspector open.

**Comment by mousemke** [comment]
updated

**Comment by mousemke** [comment]
^^ just noticed you fixed it. did it local and forgot to push

File: demon.js [hunk 4]
```diff
@@ -67,21 +93,6 @@ async function hasSuitableDate(data, xhrLink, secondShotXhrLink) {
       }
     }
 
-    if (data?.next_slot?.startsWith("2021-05")) {
```
Comment (Pita) [inline demon.js:70]: I think we can keep this, just without the `startsWith`

Comment (mousemke) [reply demon.js:70]: 👍

**Comment by Pita** [comment]
Nice, thank you! I never even expected anyone to care about this code, it was something I quickly put together in 2 evenings because I got frustrated booking an appointment

**Comment by mousemke** [comment]
it's helped me and my partner. i'm happy to contribute and i've been suggesting it to people

**Pull request merged**